pkg/vmprovider/providers/vsphere/session/session_vm_update.go

```diff
@@ -476,6 +476,9 @@ func updateConfigSpec(
 		UpdateHardwareConfigSpec(config, configSpec, &vmClassSpec)
 		UpdateConfigSpecCPUAllocation(config, configSpec, &vmClassSpec, updateArgs.MinCPUFreq)
 		UpdateConfigSpecMemoryAllocation(config, configSpec, &vmClassSpec)
+		// Only in this case, the VM's hardware version is not set to the
+		// minimum version from the VM.Spec, if any.
+		util.EnsureMinHardwareVersionInConfigSpec(configSpec, vmCtx.VM.Spec.MinHardwareVersion)
```

Comment:
We probably don't/can't do this in the update path even when VM Class as Config FSS is disabled.  Any changes of HardwareVersion on update would require the UpgradeAPI() right?